crates/dataflow-jit/src/ir/nodes/filter_map.rs

```diff
@@ -52,27 +63,88 @@ impl DataflowNode for Map {
         map(&mut self.input);
     }
 
-    fn output_kind(&self, inputs: &[StreamLayout]) -> Option<StreamKind> {
-        Some(inputs[0].kind())
-    }
-
-    fn output_stream(&self, inputs: &[StreamLayout]) -> Option<StreamLayout> {
-        Some(match inputs[0] {
-            StreamLayout::Set(_) => StreamLayout::Set(self.layout),
-            StreamLayout::Map(key_layout, _) => StreamLayout::Map(key_layout, self.layout),
-        })
-    }
+    fn output_kind(&self, _inputs: &[StreamLayout]) -> Option<StreamKind> {
+        Some(self.output_layout.kind())
+    }
+
+    fn output_stream(&self, _inputs: &[StreamLayout]) -> Option<StreamLayout> {
+        Some(self.output_layout)
+    }
+
+    fn validate(&self, inputs: &[StreamLayout], _layout_cache: &RowLayoutCache) {
```

Comment:
is panic the best way to fail validation?

Reply:
Right now it is, I need to refactor the validation system

Reply:
I would suggest filing an issue about it, this way we'll know it's planned and not an oversight.